src/RemoteInboxNotifications/Transformers/ArrayColumn.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications\Transformers;
+
+use Automattic\WooCommerce\Admin\RemoteInboxNotifications\TransformerInterface;
+use InvalidArgumentException;
+use stdClass;
+
+/**
+ * Search array value by one of its key.
+ *
+ * @package Automattic\WooCommerce\Admin\RemoteInboxNotifications\Transformers
+ */
+class ArrayColumn implements TransformerInterface {
+	/**
+	 * Search array value by one of its key.
+	 *
+	 * @param mixed         $value a value to transform.
+	 * @param stdClass|null $arguments required arguments 'key'.
+	 *
+	 * @throws InvalidArgumentException Throws when the required argument 'key' is missing.
+	 *
+	 * @return mixed
+	 */
+	public function transform( $value, stdClass $arguments = null ) {
+		if ( ! isset( $arguments->key ) ) {
```

Comment:
Again, this could be in the rule processor's `validate` function

Reply:
alternatively there could be a `validate` function on the transformer classes, and `OptionRuleProcessor` could just call into the appropriate implementation.

Reply:
That's a good suggestion. I think I will add `validate` in each Transformer class as they should protect themselves.

Reply:
That's a good point, but I think an object should protect itself again any values coming it. It's also going to be useful in case we ever want to use Transformers in other types other than Option. We wouldn't have to repeat code (calling validate)

Reply:
I agree that the transformer is the appropriate place for validation to happen, I just think it could be in a seperate method and called as part of rule validation.

> We wouldn't have to repeat code (calling validate)

The only code that would be repeated would be like (inside of `OptionRuleTransformer`:
```
foreach ( $rule->transformers as $transform_args ) {
    $transformer = TrasformerService::create_transformer( $transform_args->use );
    if ( ! $transformer->validate( $transform_args->arguments ) ) {
        return false;
    }
}
```

Then if and when this needs to be duplicated, we look at deduplicating it then.

Reply:
Looking at the actual code, I think we can do that. That wouldn't be too hard to copy and paste in case we ever need to. I will make an update soon 👍